File: .codex_tmp_spreadsheet/remove_weekly_review.mjs
import fs from "node:fs/promises";
import { FileBlob, SpreadsheetFile } from "@oai/artifact-tool";

const inputPath = "../BlueBridge_TOFU_BizDev_V1.xlsx";
const outputDir = "../outputs/weekly_review_removed";
const outputPath = `${outputDir}/BlueBridge_TOFU_BizDev_V1_no_weekly_review.xlsx`;

const input = await FileBlob.load(inputPath);
const workbook = await SpreadsheetFile.importXlsx(input);

const sheetInfo = await workbook.inspect({
  kind: "sheet",
  include: "id,name",
  maxChars: 4000,
});
console.log("Before:");
console.log(sheetInfo.ndjson);

const weeklyReview = workbook.worksheets.getItem("Weekly Review");
weeklyReview.delete();

const errors = await workbook.inspect({
  kind: "match",
  searchTerm: "#REF!|#DIV/0!|#VALUE!|#NAME\\?|#N/A",
  options: { useRegex: true, maxResults: 300 },
  summary: "final formula error scan",
  maxChars: 6000,
});
console.log("Formula errors:");
console.log(errors.ndjson);

const after = await workbook.inspect({
  kind: "sheet",
  include: "id,name",
  maxChars: 4000,
});
console.log("After:");
console.log(after.ndjson);

await fs.mkdir(outputDir, { recursive: true });
const output = await SpreadsheetFile.exportXlsx(workbook);
await output.save(outputPath);
console.log(`Saved ${outputPath}`);

File: .codex_tmp_spreadsheet/remove_weekly_review.py
from pathlib import Path

from openpyxl import load_workbook


root = Path(__file__).resolve().parents[1]
input_path = root / "BlueBridge_TOFU_BizDev_V1.xlsx"
output_dir = root / "outputs" / "weekly_review_removed"
output_path = output_dir / "BlueBridge_TOFU_BizDev_V1_no_weekly_review.xlsx"

wb = load_workbook(input_path)
print("Before:", wb.sheetnames)

if "Weekly Review" not in wb.sheetnames:
    raise SystemExit("Weekly Review sheet was not found.")

del wb["Weekly Review"]
print("After:", wb.sheetnames)

output_dir.mkdir(parents=True, exist_ok=True)
wb.save(output_path)
print(output_path)
